lucille-core/src/main/java/com/kmwllc/lucille/stage/DictionaryLookup.java

```diff
@@ -94,64 +115,83 @@ private HashMap<String, String[]> buildHashMap(String dictPath) throws StageExce
           continue;
         }
 
+        // save the first word for both single and multi-word lines
+        String word = line[0].trim();
+
         // TODO : Add log messages for when encoding errors occur so that they can be fixed
+        String[] value;
         if (line.length == 1) {
-          String word = line[0].trim();
-          if (ignoreCase) {
-            dict.put(word.toLowerCase(), new String[]{word});
-          } else {
-            dict.put(word, new String[]{word});
-          }
+          value = setOnly ? PRESENT : new String[]{word};
+        } else if (setOnly) {
+          throw new StageException(String.format("Comma separated values are not allowed when set_only=true: \"%s\" on line %d",
+              Arrays.toString(line), reader.getLinesRead()));
         } else {
           // Handle multiple payload values here.
-          String[] rest = Arrays.copyOfRange(line, 1, line.length);
-          for (int i = 0; i < rest.length; i++) {
-            rest[i] = rest[i].trim();
-          }
-          if (ignoreCase) {
-            dict.put(line[0].trim().toLowerCase(), rest);
-          } else {
-            dict.put(line[0].trim(), rest);
-          }
+          value = Arrays.stream(Arrays.copyOfRange(line, 1, line.length)).map(String::trim).toArray(String[]::new);
         }
+        // Add the word and its payload(s) to the dictionary
+        dict.put(ignoreCase ? word.toLowerCase() : word, value);
       }
     } catch (IOException e) {
       throw new StageException("Failed to read from the given file.", e);
     } catch (CsvValidationException e) {
       throw new StageException("Error validating CSV", e);
     }
-
     return dict;
   }
 
   @Override
   public Iterator<Document> processDocument(Document doc) throws StageException {
+
     for (int i = 0; i < sourceFields.size(); i++) {
       // If there is only one dest, use it. Otherwise, use the current source/dest.
       String sourceField = sourceFields.get(i);
       String destField = destFields.size() == 1 ? destFields.get(0) : destFields.get(i);
 
-      if (!doc.has(sourceField)) {
-        continue;
-      }
+      if (setOnly) {
 
-      List<String> outputValues = new ArrayList<>();
-      for (String value : doc.getStringList(sourceField)) {
-        if (ignoreCase) {
-          value = value.toLowerCase();
+        // default value is true if this is the first field or if there are multiple fields
+        // in case where we have one destination field and multiple source fields we want to retrieve the current value
+        boolean defaultValue = true;
+        if (i != 0 && destFields.size() == 1) {
+          defaultValue = doc.getBoolean(destField);
```

Comment:
If we see a false here, it looks like the value is always going to stay false, so we could break out of the loop, right? Except the updateMode could be APPEND or SKIP in which case doc.getBoolean() is not going to give us the result of the previous set lookup.
I think it we are using multiple source fields with one destination, we have to prohibit UpdateMode.APPEND from being specified, otherwise the logic doesn't work -- what do you think?